examples/DS3231_Temperature/DS3231_Temperature.ino

```diff
@@ -0,0 +1,164 @@
+#include <RtcDS3231.h>
+
+SoftwareWire myWire(SDA, SCL);
+RtcDS3231<SoftwareWire> Rtc(myWire);
+ for software wire use above */
+
+/* for normal hardware wire use below */
+#include <Wire.h> // must be included here so that Arduino library object file references work
+#include <RtcDS3231.h>
+RtcDS3231<TwoWire> Rtc(Wire);
+/* for normal hardware wire use above */
+
+
+void setup () 
+{
+    Serial.begin(57600);
+    while (!Serial) delay(250);    // Wait until Arduino Serial Monitor opens
+    Serial.println( " " );         // For ESP8266, newline after bootup message
+    
+    Serial.println(F("RtcDS3231 Misc. Temperature Tests"));
+
+    Serial.print("compiled: ");
+    Serial.print(__DATE__);
+    Serial.print( "_" );       // Underscore to distinguish from 'printDateTime()' output
+    Serial.println(__TIME__);
+
+    //--------RTC SETUP ------------
+    // if you are using ESP-01 then uncomment the line below to reset the pins to
+    // the available pins for SDA, SCL
+    // Wire.begin(0, 2); // due to limited pins, use pin 0 and 2 for SDA, SCL
+    
+    Rtc.Begin();
+
+    RtcDateTime compiled = RtcDateTime(__DATE__, __TIME__);
+    printDateTime(compiled);
+    Serial.println();
+
+    if (!Rtc.IsDateTimeValid()) 
+    {
+        // Common Causes:
+        //    1) first time you ran and the device wasn't running yet
+        //    2) the battery on the device is low or even missing
+
+        Serial.println("RTC lost confidence in the DateTime!");
+
+        // following line sets the RTC to the date & time this sketch was compiled
+        // it will also reset the valid flag internally unless the Rtc device is
+        // having an issue
+
+        Rtc.SetDateTime(compiled);
+    }
+
+    if (!Rtc.GetIsRunning())
+    {
+        Serial.println("RTC was not actively running, starting now");
+        Rtc.SetIsRunning(true);
+    }
+
+    RtcDateTime now = Rtc.GetDateTime();
+    if (now < compiled) 
+    {
+        Serial.println("RTC is older than compile time!  (Updating DateTime)");
+        Rtc.SetDateTime(compiled);
+    }
+    else if (now > compiled) 
+    {
+        Serial.println("RTC is newer than compile time. (this is expected)");
+    }
+    else if (now == compiled) 
+    {
+        Serial.println("RTC is the same as compile time! (not expected but all is fine)");
+    }
+    Serial.println();
+
+    // never assume the Rtc was last configured by you, so
+    // just clear them to your needed state
+    Rtc.Enable32kHzPin(false);
+    Rtc.SetSquareWavePin(DS3231SquareWavePin_ModeNone);
+    
+} //setup
+
+// Temperature test loop
+void loop () 
+{
+    if (!Rtc.IsDateTimeValid()) 
+    {
+        // Common Causes:
+        //    1) the battery on the device is low or even missing and the power line was disconnected
+        Serial.println("RTC lost confidence in the DateTime!");
+    }
+
+    // Force a temperature A/D conversion
+    Rtc.ForceTemperatureCompensationUpdate( true );
```

Comment:
Using the method "ForceTemperatureCompenstationUpdate()" should be avoided in most cases.  It is not needed to get an accurate temperature. 

All it does is force the chip to change how it compensates the passage of time based on temperature to be reevaluated, which it already does periodically.  Calling this is more important if you want a "VERY" accurate time in a "VERY" fast changing temperature environment.

Were you finding this a requirement to have the temperatures change at all?  This is not the case for earlier RTC chips.

Reply:
Remember that 'DS3231_Temperature.ino' is intended to test the methods included in RtcTemperature. The "ForceTemperatureCompenstationUpdate()" is included to show its used, and a rapid temperature update is 
desired here to show heating/cooling of the part.

Have added a comment stating the above to the sketch